src/allocation/quadraticVoting/template.sol

```diff
@@ -41,28 +44,51 @@ contract QVAllocationStrategy is IAllocationStrategy, Initializable {
     }
 
     function applyToPool(
-        bytes memory _data,
-        address sender
-    ) external payable returns (bytes memory) {
+        bytes memory _data
+    ) external payable override returns (bytes memory) {
         // decode data to get
         //  - identityId
         //  - applicationMetaPtr
         //  - recipientAddress
-
+        (
+            address identityId,
+            MetaPtr memory applicationMetaPtr,
+            address recipientAddress
+        ) = abi.decode(_data, (address, MetaPtr, address));
         // NOTE: custom logic if we wanted to gate applications based on EAS / registry check
- 
+        Application storage application = applications[identityId];
         // set application status to pending or reapplied
-        // add / update applications mapping
+        if (application.status == ApplicationStatus.None) {
+            application.status = ApplicationStatus.Pending;
+        } else if (application.status == ApplicationStatus.Rejected) {
+            // todo: will it ever be rejected at this point?
+        } else {
+            // todo: figure the reapplied out
+            // application.status = ApplicationStatus.Reapplied;
+        }
```

Comment:
^ i think we could simplify this and just say if status is NONE -> it's pending 
Else Reapplied. Will worry about actual toggle logic when we build this out